performance/swiftinterop/swiftSrc/benchmarks.swift

```diff
@@ -0,0 +1,170 @@
+        for _ in 0...BENCHMARK_SIZE {
+            multigraph.addEdge(from: randomInt(border: BENCHMARK_SIZE), to: randomInt(border: BENCHMARK_SIZE), cost: SimpleCost(cost: randomInt(border: BENCHMARK_SIZE)))
+        }
+    }
+
+    func fillCityMap() {
+        for _ in 0...BENCHMARK_SIZE {
+            cityMap.addRoute(from: randomPlace(), to: randomPlace(), cost: randomRouteCost())
+        }
+    }
+
+    func searchRoutesInSwiftMultigraph() {
+        for _ in 0...BENCHMARK_SIZE {
+            var result = multigraph.searchRoutesWithLimits(start: randomInt(border: BENCHMARK_SIZE), finish: randomInt(border: BENCHMARK_SIZE), limits: SimpleCost(cost: randomInt(border: BENCHMARK_SIZE)))
+            var count = result.map{ $0.count }.reduce(0, +)
+        }
+    }
+
+    func searchTravelRoutes() {
+        for _ in 0...BENCHMARK_SIZE {
+            var result = cityMap.getRoutes(start: randomPlace(), finish: randomPlace(), limits: randomRouteCost())
+            var totalCost = result.flatMap{ $0 }.map { $0.cost.moneyCost }.reduce(0, +)
+        }
+
+    }
+
+    func availableTransportOnMap() {
+        for _ in 0...BENCHMARK_SIZE {
+            Set(cityMap.allRoutes.map { (cityMap.getRouteById(id: $0.id).cost as! RouteCost).transport })
+        }
+    }
+
+    func allPlacesMapedByInterests() {
+        for _ in 0...BENCHMARK_SIZE {
+            let placesByInterests = cityMap.allPlaces.reduce([Interest: Array<Place>]()) { (dict, place) -> [Interest: Array<Place>] in
+                var dict = dict
+                if (dict[place.interestCategory] != nil) {
+                    dict[place.interestCategory]?.append(place)
+                } else {
+                    dict[place.interestCategory] = [place]
+                }
+                return dict
+            }
+        }
+    }
+
+    func getAllPlacesWithStraightRoutesTo() {
+        for _ in 0...BENCHMARK_SIZE {
+            let start = cityMap.allPlaces.randomElement()!
+            var availableRoutes = cityMap.getAllStraightRoutesFrom(place: start)
+            var availablePlaces = availableRoutes.map { $0.to }
+        }
+    }
+
+    func goToAllAvailablePlaces() {
+        for _ in 0...BENCHMARK_SIZE {
+            let start = cityMap.allPlaces.randomElement()!
+            var availableRoutes = cityMap.getAllStraightRoutesFrom(place: start)
+            availableRoutes.map { 2 * $0.cost.moneyCost }
+        }
+    }
+
+    func removeVertexAndEdgesSwiftMultigraph() {
+        var multigraphCopy = multigraph.copy() as! Multigraph
+        var edges = multigraphCopy.allEdges
+        while (!edges.isEmpty) {
+            multigraphCopy.removeEdge(id: UInt32(edges.randomElement()!))
+            let vertexes = multigraphCopy.allVertexes
+            multigraphCopy.removeVertex(vertex: vertexes.randomElement()!)
+            edges = multigraphCopy.allEdges
+        }
+    }
+
+    func removePlacesAndRoutes() {
+        var cityMapCopy = cityMap.copy() as! CityMap
```

Comment:
What do you expect to get by calling `cityMap.copy`?

Reply:
Deep copy of object

Reply:
Not sure if `cityMap.copy()` creates deep copy.